src/org/zaproxy/zap/extension/httpsinfo/ExtensionHttpsInfo.java

```diff
@@ -80,8 +82,9 @@ public void unload() {
 		unloaded = true;
 
 		while (!dialogues.isEmpty()) {
-			// When disposed the dialogue is removed from the list
-			dialogues.get(0).dispose();
+			for (HttpsInfoDialog dialog : dialogues) {
```

Comment:
Ugh... talk about reading what you want to read. I remember looking at that when I fist started all these changes and thinking "if not empty dispose of the first one. What about the rest?" Totally missing that it was a while not an if.... will restore the previous behaviour.